src/test/assembly/stack-probes.rs

```diff
@@ -0,0 +1,48 @@
+// min-llvm-version: 11.0
+// revisions: x86_64 i686
+// assembly-output: emit-asm
+//[x86_64] compile-flags: --target x86_64-unknown-linux-gnu
+//[i686] compile-flags: --target i686-unknown-linux-gnu
+// compile-flags: -C llvm-args=--x86-asm-syntax=intel
+
+#![feature(no_core, lang_items)]
+#![crate_type = "lib"]
+#![no_core]
+
+#[lang = "sized"]
+trait Sized {}
+#[lang = "copy"]
+trait Copy {}
+
+// Check that inline-asm stack probes are generated correctly.
+// When stack usage is relatively small, the probing loop will be unrolled.
+
+// CHECK-LABEL: small_stack_probe:
+#[no_mangle]
+pub fn small_stack_probe(a: [u8; 8192], f: fn([u8; 8192])) {
+    // x86_64: sub rsp, 4096
+    // x86_64-NEXT: mov qword ptr [rsp], 0
+    // x86_64-NEXT: sub rsp, 4096
+    // x86_64-NEXT: mov qword ptr [rsp], 0
+    // i686: sub esp, 4096
+    // i686-NEXT: mov dword ptr [esp], 0
+    // i686-NEXT: sub esp, 4096
+    // i686-NEXT: mov dword ptr [esp], 0
+    let a = a;
+    f(a);
+}
```

Comment:
I'm unsure if this test is too fragile...it would probably be legal to optimize out the copy. (And beyond that, do we even want to test the specific instructions that LLVM generates for probes?)

Reply:
It may be better to create the array directly in this stack frame -- e.g. take a single `u8` as input and create the array from that.

As for the assembly, I don't think we should try to match the exact output. We could `CHECK-NOT` against the `__rust_probestack` symbol, and/or loosely `CHECK` for the page size step 4096, and that's probably good enough to assume some kind of probing is there.